### PR TITLE
Add action to apply updates in /docs to /versioned_docs/version-XYZ

### PR DESCRIPTION
**Pull request by btat**
Depends on #665.

With the introduction of having a `versioned_docs` entry to represent the latest version, we'll have increased overhead of making sure `/docs` and `versioned_docs/version-<latest>` ( `version-2.7` in this case) are in sync. With this action, when a PR is made against `main`, the corresponding changes will be applied against `versioned_docs/version-<latest>` as well and committed to the same PR.

### CODE REVIEW

File: .github/workflows/test-deploy.yml [hunk 1]
```diff
@@ -16,6 +16,24 @@ jobs:
...
+        if: steps.changed-files-specific.outputs.any_changed == 'true'
+        run: |
+          cp -r ./docs/* ./versioned_docs/version-2.7/
+          git config user.name "GitHub Actions Bot"
+          git config user.email "<>"
+          git config --global --add --bool push.autoSetupRemote true
+          gh pr checkout ${{ github.event.pull_request.number }}
+          git add ./versioned_docs/version-2.7/
+          git commit -m "Apply change(s) to versioned_docs/version-2.7"
+          git push
```
[the remark this answers is not in the document]
Comment (martyav) [reply .github/workflows/test-deploy.yml:36]: Looks like we can use something like the following (from https://github.com/tj-actions/changed-files#usage) to target particular folders:

```
- name: Get changed files in the docs folder
        id: changed-files-specific
        uses: tj-actions/changed-files@v36
        with:
          files: docs/*.{js,html}  # Alternatively using: `docs/**` or `docs`
```

**Review by martyav — COMMENTED** [review]
I left some deleted comments but I see now this will only trigger on changes specifically in the /docs folder. My only remaining concern is that editors might shoot themselves in the foot by editing /version_2.7 instead of /docs and then wondering where their changes went. While we can warn against doing that, it would be better if we had some mechanism preventing it, whether that's an additional action keeping the pages in sync, or something that keeps changes from being pushed if they touch /2.7 pages directly

**Comment by martyav** [comment]
Oh -- we also need to keep the sidebars in sync.

**Comment by btat** [comment]
> Oh -- we also need to keep the sidebars in sync.

I thought about that case and poked around for a bit, but it wasn't obvious how Docusaurus does the sidebar js -> json conversion. But I don't think this should be a blocker since it's not a regular occurrence that a sidebar affecting change is made  (in which case there would be other things to update) and we'll still need to review such changes anyways.

**Comment by btat** [comment]
PR updated. 

As discussed, the case where `/docs` and `versioned_docs/version-<XYZ>` are both updated is out of scope for this PR. When both directories have updated files:
- The same file(s) may be modified in both directories in which case manual review is needed.
- Both directories have modified files, but files are only modified in one of the directories. This will can be a future improvement.

**Review by martyav — APPROVED** [review]

**Pull request merged**